### ISSUE TITLE
Default values for environment variables.

### ISSUE THREAD

**Issue by VVasyaev** [robotframework/robotframework#3382]
Hello!

All of my projects uses a bunch of env vars for configuration.

Sometimes these needs to have some defaults.

As a workaround i made a keyword with `OperatingSystem.Get Environment Variable` and try to load the env var there or put default. The cons of it is -- any keyword, which uses my env var's value need to get that 'loading' keyword running. It seems to be silly.

I'd like to have some default value feature in the `%{}` structure.

For example:
```
*** Variables ***
${var_uses_everywhere}    %{MY_ENV_VAR:default_value_if_missing}
```
or
`%{MY_ENV_VAR:-default_value_if_missing}`
to mimic bash `${a:-b}` structure.

It seems not so hard to implement such behavior.

**Comment by pekkaklarck** [comment]
Although it shouldn't be overly complicated to make sure all needed environment variables are set before execution, I can see this could sometimes be useful and make configuration easier. I got a feeling that `:` is actually allowed in environment variable name (shell may not like it), but `=` shouldn't be so `%{VAR=default}` would be a bit better.

Are you interested to implement this and provide a PR? I can help if needed.

**Comment by VVasyaev** [comment]
One of my project consists of lot of services, each of them running the same RF-tests repository (all of them have common keywords, resources, libraries, etc.) with different tags which helps to start the correct tests. The main issue is -- each service can be started on different envs and have lot of configuration differences (here are several dev teams creating these services) -- so defaults is very important for me.

As for PR -- i would like to try! Anyway, you can always say what i did wrong in PR. However, i didn't contribute in RF code earlier so if you have some suggestions -- i'd appreciate!

**Comment by VVasyaev** [comment]
@pekkaklarck pls see https://github.com/robotframework/robotframework/pull/3384 PR
i'd also added two test-cases

**Comment by pekkaklarck** [comment]
Yeah, there definitely are use cases where this would help. Robot's own acceptance test runner sets a Python interpreter and OS specific `TEMPDIR` environment variable, and sometimes when running tests directly, without the runner, you need to explicitly set it as well. We could change the current usages of `%{TEMPDIR}` to `%{TEMPDIR=${TEMPDIR}}` to use the OS default temporary directory in these cases.

Implementing this would actually be pretty simple. The code related to finding environment variables is [pretty simple](https://github.com/robotframework/robotframework/blob/master/src/robot/variables/finders.py#L165) and editing it to split the possible default value from the name and returning it if there's no match shouldn't be hard either. You'd need to also create some acceptance tests and update docs.

To get started, see the [contributing guidelines](https://github.com/robotframework/robotframework/blob/master/CONTRIBUTING.rst). If you have questions, don't hesitate to ask here or on the `#devel` channel in our Slack.

**Comment by pekkaklarck** [comment]
Didn't see you earlier comment or the PR when I wrote my previous comment. Good start with the PR but some changes are needed.

**Comment by VVasyaev** [comment]
@pekkaklarck pls review #3384 one more time...

**Comment by pekkaklarck** [comment]
This was implemented by the great PR #3384 by @VVasyaev!
